File: src/dataset/go/binary_search_Exercism_go_drapala/claude/ClaudeSonnet4_binary_search_v1.go
package binarysearch

func SearchInts(list []int, key int) int {
	l, r := 0, len(list)-1
	
	for l <= r {
		m := l + (r-l)>>1
		val := list[m]
		
		if val < key {
			l = m + 1
		} else if val > key {
			r = m - 1
		} else {
			return m
		}
	}
	return -1
}

File: src/dataset/c/binary_search_Exercism_HeitorMP/claude/ClaudeSonnet4_binary_search_v1.c
#include "binary_search.h"

int *binary_search(int value, const int *arr, size_t length)
{
    if (!arr || length == 0)
        return NULL;
    
    size_t start = 0;
    size_t end = length;
    
    while (start < end)
    {
        size_t mid = start + ((end - start) >> 1);
        int mid_value = arr[mid];
        
        if (mid_value == value)
            return (int *)&arr[mid];
        else if (mid_value < value)
            start = mid + 1;
        else
            end = mid;
    }
    return NULL;
}